src/js/components/SingleImage.vue

```diff
@@ -36,73 +36,192 @@
   },
   data() {
     return {
       imageElements: [],
       selectedImageUrls: [],
       filenames: [],
-      navPanel: 'seadragonNavPanel',
+      masterImage: {},
+      imageShrink: 2,
+      artefact: undefined,
+      artName: '',
       zoom: 0.5,
       scale: 0.2,
       selectedImage: undefined,
+      viewMode: 'ART',
+      drawingMode: 'draw',
+      brushCursorSize: 20,
+      clipMask: '',
+      firstClipMask: '',
+      clippingOn: false,
+      lock: true,
     }
   },
   methods: {
+    fetchImages(id) {
+      this.$post('resources/cgi-bin/scrollery-cgi.pl', {
+        transaction: 'imagesOfInstFragments',
+        id: id,
+      })
+      .then(res => {
+        if (res.status === 200 && res.data.results) {
+          this.imageElements = res.data.results
+          this.filenames = []
+          res.data.results.forEach((result) => {
+            if (result.is_master >>> 0) {
+              result.visible = true
+              this.masterImage = result
+            }
+            result.opacity = 1.0
+            this.filenames.push(result)
+          })
+          this.masterImage = this.masterImage ? this.masterImage : res.data.results[0] // Is the ternary really necessary?
+          // this.clipMask = undefined
+          this.clipMask = this.clipMask ? this.clipMask : this.fullImageMask // this would catch edge cases, but may be unnecessary
+        }
+      })
+    },
+    getArtefactMask() {
+      this.$post('resources/cgi-bin/scrollery-cgi.pl', {
+        transaction: 'getArtefactMask',
+        artID: this.artefact,
+      })
+      .then(res => {
+        if (res.status === 200 && res.data.results[0]) {
+          this.firstClipMask = this.clipMask = wktPolygonToSvg(res.data.results[0].poly)
+        }
+      })
+    },
+    setClipMask(mask) {
+      this.lock = true
+      this.clipMask = mask
+      if (this.artefact === 'new') {
+        this.$post('resources/cgi-bin/scrollery-cgi.pl', {
+          transaction: 'newArtefact',
+          image_id: this.$route.params.imageID,
+          region_in_master_image: svgPolygonToWKT(mask),
+          name: this.artName,
+          scroll_id: this.$route.params.scrollID,
+          version_id: this.$route.params.scrollVersionID,
+        })
+        .then(res => {
+            if (res.status === 200 && res.data.returned_info) {
+	            this.$router.push({
+                name: 'workbenchAddress',
+                params: {
+                  scrollID: this.$route.params.scrollID,
+                  scrollVersionID: this.$route.params.scrollVersionID,
+                  colID: this.$route.params.colID ? this.$route.params.colID : '~',
+                  imageID: this.$route.params.imageID ? this.$route.params.imageID : '~',
+                  artID: res.data.returned_info
+                }
+              })
+              this.lock = false
+            }
+        })
+      } else {
+	      this.$post('resources/cgi-bin/scrollery-cgi.pl', {
```

Comment:
nit: looks like the alignment is off here.

Reply:
Nice catch!  IntelliJ can't seem to understand the formatting I want.  Do you have some sort of way in VSCode to auto format the way we like it?

Reply:
In VSCode, I just highlight the portion I want to format and right click > Format Selection (there's a hotkey too...). You may need to adjust your user settings that VSCode knows about how to format (e.g., the number of spaces for indent). It's usually pretty good once you tweak the settings.

I'll look at at a prettier integration as well so we can get some auto-enforment going on too.